### PR TITLE
Add trait PlayerGLContext

### PR DESCRIPTION
**Pull request by ceyusa**
PlayerGLContext will provide the GL context paratemers required to
create a Render object. If those parameters are supported by the
Render, it will setup a GL rendering pipeline and will handle the
frames in form of GL Textures.

The PlayerGLContext is passed to the player at its instantiation.

### CODE REVIEW

**Comment by ceyusa** [comment]
@ferjm r?

**Comment by ceyusa** [comment]
this needs a modification in servo, since the player instantiation changed

File: player/context.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//! `PlayerGLContext` is a trait to be used to pass the GL context for
+//! rendering purposes.
+//!
+//! The current consumer of this trait is the GL rendering mechanism
+//! for the GStreamer backend.
+//!
+//! The client application should implement this trait and pass the
+//! trait object to its `player` intsance.
```
Comment (ferjm) [inline player/context.rs]: typo: instance

**Pull request merged**